File: src/use-cases/implementations/get-nearest-gyms.ts
import { inject, injectable } from "inversify";
import { Repositories } from "../../common/dependency-injection";
import { LocationEntity } from "../../entities";
import { IGymRepository } from "../../repositories/interfaces";
import { IGetNearestGymsUseCase } from "../interfaces";

@injectable()
export class GetNearestGymsUseCase implements IGetNearestGymsUseCase {
  gymRepo: IGymRepository;

  constructor(@inject(Repositories.gym) gymRepo: IGymRepository) {
    this.gymRepo = gymRepo;
  }

  execute(location: LocationEntity, distance: number, amount?: number) {
    return this.gymRepo.getNearest(location, distance, amount);
  }
}

File: lib/src/use-cases/implementations/get_nearest_gyms.dart
import '../../entities/entities.dart';
import '../../repositories/repositories.dart';
import '../interfaces/interfaces.dart';

class GetNearestGymsUseCase implements IGetNearestGymsUseCase {
  @override
  Future<List<GymEntity>> get({
    required LocationEntity location,
    required int maxDistance,
    int? limit,
  }) =>
      gymRepo.getNearestGyms(
        location: location,
        distance: maxDistance,
        amount: limit,
      );
}
